File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        arrSet = set(nums1)
        result = []

        for num in nums2:
            if num in arrSet:
                result.append(num)
                arrSet.remove(num)
                
        return result

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    const set = new Set(nums1);
    const result: number[] = [];

    for (let num of nums2) {
        if (set.has(num)) {
            result.push(num);
            set.delete(num);
        }
    }
    return result;
};